cmd/serve.go

```diff
@@ -439,14 +393,22 @@ func flood(ctx context.Context, remote string) (info perf.Perf, err error) {
 		threads := steps[i].threads
 
 		if info, err = doFlood(ctx, remote, size, threads); err != nil {
-			if err == networkOverloaded {
-				continue
+			if ctx.Err() != nil {
+				return info, err
```

Comment:
do you want to return ctx.Err() or the one returned by doFlood?

Reply:
if it failed because ctx cancelled, then the doFlood func will throw context cancelled. doFlood error is what we care about